Cleanup and semantic changes

olorin opened the pull request:
Reworked a lot of things.
- refactored all the things to reduce code duplication and enhance testability
- changed siphash to use a null-byte string for consistency with everything else we use siphash for
- don't write out duplicate consolidated streams when we don't need to
- don't include flavor ID in the identifier of all metrics
- cut down sourcedicts substantially, don't include 
  - flattened resource metadata 
  - flavor data
- add more consolidated (pollster) instance metrics
  - instance_vcpus
  - instance_ram

olorin committed:
This isn't used anymore

olorin committed:
Use null bytes rather than the '0' character as the siphash key, for consistency

olorin committed:
More informative names, add some explanatory docstrings

olorin committed:
Don't write out consolidated duplicates of everything

olorin committed:
Bump version

olorin committed:
We changed that param name

olorin committed:
Update tests for siphash changes

olorin committed:
Refactor source dict building to reduce code duplication

olorin commented:
Don't merge just yet, I'll add a couple more changes before rolling out.

olorin committed:
Rename some things in line with PEP 8

olorin committed:
consolidated doesn't really make sense on its own as a module

olorin committed:
Factor out some more code duplication

olorin committed:
Remove flavor ID from identifier + refactoring

We don't want new addresses for all instance-related metrics every time
an instance gets resized.

Also factored out common elements of ID generation.

olorin committed:
Don't try to import modules which don't exist

oswynb commented on ceilometer_publisher_vaultaire/process.py:
```diff
@@ -103,7 +87,45 @@ def process_raw(sample):
+
+    return sourcedict
+
+def get_address(sample, name, consolidated=False):
```
Why is name explicitly separated from the rest of the sample?

olorin replied:
Because sometimes we want to write metrics with names which are different from those in the Ceilometer message (e.g., for instance VCPU and memory allocations).

oswynb replied:
ahh cool, makes sense.

oswynb commented on ceilometer_publisher_vaultaire/process.py:
```diff
@@ -103,7 +87,45 @@ def process_raw(sample):
+    return Marquise.hash_identifier(identifier)
+
 def process_consolidated_pollster(sample):
+    frames = []
```
what is this used for?

olorin replied:
Nothing yet, slipped into the commit when I got ahead of myself earlier. I'll get rid of it for now.

oswynb commented on ceilometer_publisher_vaultaire/process.py:
```diff
@@ -182,54 +198,22 @@ def process_consolidated_event(sample):
+        payload = construct_payload(metadata["event_type"], metadata.get("status",""), volume_to_raw_payload(sample["volume"]))
     elif name.startswith("ip.floating"):
-        payload = consolidated.constructPayload(metadata["event_type"], "", 1)
+        payload = construct_payload(metadata["event_type"], "", 1)
```
Should probably be RAW_PAYLOAD_IP_ALLOC instead of 1 like in pollster

olorin replied:
Agreed, fixed.

olorin committed:
Factor out more duplicate logic

olorin committed:
More deduplication of logic

olorin committed:
More cleanup

olorin committed:
Better names for some things

olorin committed:
Fail loudly if consolidated processing breaks

Rather than silently continuing; we want to know if this fails.

olorin committed:
Be more selective about what ends up in our sourcedicts

 - no flattened resource_metadata
 - strip out flavor ID

olorin committed:
Don't try and process consolidated events unless we know we want to

olorin committed:
Refactor and start splitting up monolithic process functions

olorin committed:
Break up event consolidation into type-specific functions

olorin committed:
Remove obsolete comments

olorin committed:
That if statement is now redundant

olorin committed:
Probably shouldn't worry about case here

olorin committed:
Give consolidated instance pollsters a more descriptive name

olorin committed:
Add CPU and RAM metrics derived from instance pollsters

olorin committed:
Rename counter_{name,type} to metric_{name,type}

They're not all counters so metric_ seems to make more sense to me; may
as well do it here with all the other breaking changes.

olorin committed:
This function doesn't really do anything and makes the code harder to read

olorin committed:
flavor is inside resource_metadata

olorin committed:
Merge pull request #27 from anchor/rename-counter-tags

Rename counter_{name,type} to metric_{name,type}

olorin committed:
Update tests for instance pollster changes

olorin committed:
Merge branch 'sio-wip' of github.com:anchor/ceilometer-publisher-vaultaire into sio-wip

olorin committed:
Update tests for renamed counter -> metric

olorin committed:
Merge pull request #26 from anchor/unique-consolidated-names

Give consolidated instance pollsters a more descriptive name

olorin committed:
counter* needs to be renamed here too

olorin committed:
Update tests for sourcedict changes

olorin committed:
Whitespace

olorin committed:
Tests for instance RAM and vcpus

olorin committed:
Call new test

olorin committed:
Cleanup

olorin committed:
Get rid of more extraneous keys

barneydesmond committed:
Some docstring cleanup according to PEP-257

barneydesmond committed:
whitespace tidying

barneydesmond committed:
Ooh we've got a function to do that

barneydesmond committed:
Lint out some code and variables that aren't used
